manifests/key.pp

```diff
@@ -15,7 +15,7 @@
     command     => "/usr/sbin/dnssec-keygen -a HMAC-MD5 -r /dev/urandom -b 128 -n USER ${name}",
     cwd         => "${cfg_dir}/bind.keys.d",
     require     => [
-      Package['dnssec-tools','bind9'],
+      Package['dnssec-tools',$package],
```

Comment:
My apologies, when I was preparing a clean branch for the pull request I neglected to include the $package = $dns::server::params::package line in the file, so you're right, $package would have been undef. I pushed a new commit that fully qualifies it, opting to just fully qualify it in the Package require.

Regarding just requiring dnssec-tools, that is probably a better choice. I was just doing the minimum necessary to fix the bug and get it working on RedHat while attempting to preserve the function of the original code. If you'd rather go that route, I'll push a commit that removes the requirement for the bind package entirely.